src/solvers/petsc_dm_wrapper.C

```diff
@@ -91,6 +91,11 @@ void PetscDMWrapper::build_section( const System & system, PetscSection & sectio
   // Final setup of PetscSection
   ierr = PetscSectionSetUp(section);CHKERRABORT(system.comm().get(),ierr);
 
+  // Sanity checking at least that total n_dofs match
+#ifndef NDEBUG
+  this->check_section_n_dofs_match(system, section);
```

Comment:
Should this just return total_n_dofs and then you can save 2 lines by wrapping it in libmesh_assert_equal_to()?

Reply:
Good point. Comment is actually wrong, it should be `n_local_dofs`, but I can make the suggested change, that will be better, thanks.